src/connectors/smart-sort/connectSmartSort.ts

```diff
@@ -0,0 +1,126 @@
+import { Connector } from '../../types';
+import {
+  createDocumentationMessageGenerator,
+  mergeSearchParameters,
+  noop,
+} from '../../lib/utils';
+import algoliasearchHelper from 'algoliasearch-helper';
+
+export type SmartSortConnectorParams = {
+  relevancyStrictness?: number;
+};
+
+export type SmartSortRendererOptions = {
+  isSmartSorted: boolean;
+  refine: (relevancyStrictness: number) => void;
+};
+
+const withUsage = createDocumentationMessageGenerator({
+  name: 'smartSort',
+  connector: true,
+});
+
+export type SmartSortConnector = Connector<
+  SmartSortRendererOptions,
+  SmartSortConnectorParams
+>;
+
+const connectSmartSort: SmartSortConnector = function connectSmartSort(
+  renderFn = noop,
+  unmountFn = noop
+) {
+  return widgetParams => {
+    if (
+      typeof widgetParams.relevancyStrictness !== 'undefined' &&
+      (widgetParams.relevancyStrictness < 0 ||
+        widgetParams.relevancyStrictness > 100)
+    ) {
+      throw new Error(
+        withUsage('The `relevancyStrictness` option must be between 0 and 100.')
+      );
+    }
+
+    return {
+      $$type: 'ais.smartSort',
+
+      init(initOptions) {
+        const { instantSearchInstance } = initOptions;
+        renderFn(
+          {
+            ...this.getWidgetRenderState(initOptions),
+            instantSearchInstance,
+          },
+          true
+        );
+      },
+
+      render(renderOptions) {
+        const { instantSearchInstance } = renderOptions;
+
+        renderFn(
+          {
+            ...this.getWidgetRenderState(renderOptions),
+            instantSearchInstance,
+          },
+          false
+        );
+      },
+
+      dispose({ state }) {
+        unmountFn();
+
+        return state.setQueryParameter('relevancyStrictness', undefined);
+      },
+
+      getRenderState(renderState, renderOptions) {
+        return {
+          ...renderState,
+          smartSort: this.getWidgetRenderState(renderOptions),
+        };
+      },
+
+      getWidgetRenderState({ results, helper }) {
+        const { appliedRelevancyStrictness } = results || {};
+
+        return {
+          isSmartSorted:
+            typeof appliedRelevancyStrictness !== 'undefined' &&
+            appliedRelevancyStrictness > 0 &&
+            appliedRelevancyStrictness <= 100,
+          refine: relevancyStrictness => {
```

Comment:
this function reference isn't stable, meaning it won't be the same between renders. You can set it once like in search box and other widgets

Reply:
https://github.com/algolia/instantsearch.js/pull/4648/commits/09af204c52a8d328b63a9e09e8ed42872ae9eb19